src/bud/mode.ts

```diff
@@ -0,0 +1,46 @@
+import {argv} from 'yargs'
+import {env} from './state/env'
+
+/**
+ * Fallback env
+ */
+const envFallback: string = 'production'
+
+/**
+ * specified via CLI arg
+ */
+const envArgument: any = argv.env
+
+/**
+ * specified via project .env
+ */
+const envProject: string = env?.APP_ENV || envFallback
```

Comment:
It could be more suitable to use a null-coalesce (`??`) here instead of the logical OR (`||`).